codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -97,18 +98,23 @@ class FileCodesignVisitor {
     _initialize();
 
     await Future.value(null);
-    logger!.info('Codesigned all binaries in ${tempDir.path}');
+    log.info('Codesigned all binaries in ${tempDir.path}');
 
     await tempDir.delete(recursive: true);
   }
 
-  /// Visit a [Directory] type while examining the file system extracted from an
-  /// artifact.
+  /// Visit a [Directory] type while examining the file system extracted from an artifact.
   Future<void> visitDirectory({
     required Directory directory,
     required String entitlementParentPath,
   }) async {
-    logger!.info('Visiting directory ${directory.absolute.path}\n');
+    log.info('Visiting directory ${directory.absolute.path}\n');
+    if (directoriesVisited.contains(directory.absolute.path)) {
+      throw CodesignException(
```

Comment:
I don't think failing here is the right approach as its common for symlinks to accidentally point back to the top of a parent node. Since this would be a build failure, and the codesign script has no immediate rememdies, it should log a warning and treat it as a no-op.

Reply:
I agree this case would be orthogonal to codesigning, but wouldn't this be an error while building the archives, and thus be something we want to catch?